editor/src/dev/dev.clj

```diff
@@ -754,3 +773,266 @@
+        col-doc puget.color/document
+        col-txt puget.color/text]
+    (letfn [(cls-tag [printer ^Class class]
+              (col-doc printer :class-name [:span "#" (.getSimpleName class)]))
+
+            (cls-tag-doc [printer ^Class class document]
+              [:group
+               (cls-tag printer class)
+               document])
+
+            (object-data-pprint-handler [printer-opts object->value printer object]
+              (cls-tag-doc
+                (cond-> pretty-printer printer-opts (merge printer-opts))
+                (class object)
+                (fmt-doc printer (object->value object))))]
+
+      (let [editor-pprint-handlers
+            {(namespaced-class-symbol AABB)
+             (letfn [(aabb->value [^AABB aabb]
+                       {:min (math/vecmath->clj (.min aabb))
+                        :max (math/vecmath->clj (.max aabb))})]
+               (partial object-data-pprint-handler {:sort-keys false} aabb->value))
+
+             (namespaced-class-symbol Cursor)
+             (partial object-data-pprint-handler nil code.data/cursor-print-data)
+
+             (namespaced-class-symbol CursorRange)
+             (partial object-data-pprint-handler nil code.data/cursor-range-print-data)
+
+             (namespaced-class-symbol VertexBuffer)
+             (partial object-data-pprint-handler nil vertex-buffer-print-data)}
+
+            graph-pprint-handlers
+            {(namespaced-class-symbol Arc)
+             (partial object-data-pprint-handler nil (juxt gt/source-id gt/source-label gt/target-id gt/target-label))
+
+             (namespaced-class-symbol Endpoint)
+             (partial object-data-pprint-handler nil (juxt g/endpoint-node-id g/endpoint-label))}
+
+            java-pprint-handlers
+            {(namespaced-class-symbol Class)
+             (fn class-pprint-handler [printer ^Class class]
+               (col-txt printer :class-name (.getName class)))}
+
+            resource-pprint-handlers
+            (letfn [(project-resource->value [resource]
+                      [(resource/proj-path resource)])
+
+                    (project-resource-pprint-handler [printer resource]
+                      (object-data-pprint-handler nil project-resource->value printer resource))]
+
+              {(namespaced-class-symbol FileResource)
+               project-resource-pprint-handler
+
+               (namespaced-class-symbol MemoryResource)
+               (fn memory-resource-pprint-handler [printer resource]
+                 (object-data-pprint-handler nil (comp vector resource/ext) printer resource))
+
+               (namespaced-class-symbol ZipResource)
+               project-resource-pprint-handler})
+
+            vecmath-pprint-handlers
+            (letfn [(vecmath-tuple-pprint-handler [printer vecmath-value]
+                      (object-data-pprint-handler nil math/vecmath->clj printer vecmath-value))
+
+                    (vecmath-matrix-pprint-handler [^long dim printer matrix]
```

Comment:
The `pprint` handler we use for matrices is kind of complex because it tries to format the output in a readable way. It takes care to truncate excessive decimal places, ensure the columns are aligned, and colorize zero values so you can get an approximate idea of how they will transform things.